boards/intel/index.rst

```diff
@@ -7,4 +7,4 @@ Intel Corporation
    :maxdepth: 1
    :glob:
 
-   **/*
+   [!common]*/**/*
```

Comment:
does this really work? AFAIK glob `[]` is a range, so it won't pick any folder starting with `c`, `o`, `m` or `n` here.

Reply:
well, at least currently it works this way
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/x86/index.rst

Reply:
and with this change
https://builds.zephyrproject.io/zephyr/pr/69075/docs/boards/intel/index.html

Reply:
maybe it works by chance, but this is still wrong

Reply:
```
~/zephyrproject/zephyr/boards/intel$ ls -la [!common]*/**/*
-rw-r--r-- 1 rootroot rootroot 1892 Feb 16 10:52 intel_adl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2416 Feb 16 10:53 intel_ehl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2272 Feb 16 10:52 intel_rpl/doc/index.rst
~/zephyrproject/zephyr/boards/intel$ ls -la */**/*
-rwxr-xr-x 1 rootroot rootroot 1487 Feb 15 12:22 common/scripts/build_grub.sh
-rw-r--r-- 1 rootroot rootroot 1892 Feb 16 10:52 intel_adl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2416 Feb 16 10:53 intel_ehl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2272 Feb 16 10:52 intel_rpl/doc/index.rst
```

Reply:
```
$ cd boards/indel
$ mkdir -p cintel_test/doc
$ touch cintel_test/doc/index.rst
$ ls -la */**/*
-rw-r--r--. 1 gmarull gmarull    0 Feb 16 11:49 cintel_test/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 1904 Feb 16 11:47 intel_adl/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 2440 Feb 16 11:47 intel_ehl/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 2284 Feb 16 11:47 intel_rpl/doc/index.rst
$ ls -la [!common]*/**/*
-rw-r--r--. 1 gmarull gmarull 1904 Feb 16 11:47 intel_adl/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 2440 Feb 16 11:47 intel_ehl/doc/index.rst
-rw-r--r--. 1 gmarull gmarull 2284 Feb 16 11:47 intel_rpl/doc/index.rst
### OOOOPS cintel_test missing
```

Reply:
```
~/zephyrproject/zephyr/boards/intel$ ls -la */**/index.rst
-rw-r--r-- 1 rootroot rootroot    0 Feb 16 12:16 cintel/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 1892 Feb 16 10:52 intel_adl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2416 Feb 16 10:53 intel_ehl/doc/index.rst
-rw-r--r-- 1 rootroot rootroot 2272 Feb 16 10:52 intel_rpl/doc/index.rst
```